### PR TITLE
remove noop option `gc:v2`

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/RFCs/issues/437

### CODE REVIEW

File: lib/system/sysstr.nim [hunk 1]
```diff
@@ -304,20 +304,10 @@ proc setLengthSeq(seq: PGenericSeq, elemSize, elemAlign, newLen: int): PGenericS
     when not defined(boehmGC) and not defined(nogc) and
          not defined(gcMarkAndSweep) and not defined(gogc) and
          not defined(gcRegions):
-      when false: # deadcode: was used by `compileOption("gc", "v2")`
```
Comment (Araq) [inline lib/system/sysstr.nim:307]: Looks like an unrelated improvement that I don't understand. Please explain.

Comment (ringabout) [reply lib/system/sysstr.nim:307]: I removed the `when false` branch which is marked as dead code and used by gc:v2 => https://github.com/nim-lang/Nim/blob/f561afae419212e6eeb33f37f834aa7e9d4b9d6f/lib/system/sysstr.nim#L307 which shouldn't have actual effects. I didn't touch the other branch.

Comment (ringabout) [reply lib/system/sysstr.nim:307]: The GitHub diff cannot show the actual changes reliably.
